util/src/main/java/com/google/cloud/hadoop/util/authorization/StorageRequestSummary.java

```diff
@@ -0,0 +1,81 @@
+package com.google.cloud.hadoop.util.authorization;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Encapsulation of StorageRequest.
+ *
+ * <p>The encapsulation only contains resource related information. Including: action, GCS bucket
+ * name and object path.
+ */
+public class StorageRequestSummary {
+
+  private final ActionType actionType;
+  private final List<GcsStorage> resources;
+
+  public StorageRequestSummary(ActionType actionType, List<GcsStorage> resources) {
+    this.actionType = actionType;
+    this.resources = resources;
+  }
+
+  public ActionType getActionType() {
+    return actionType;
+  }
+
+  public List<GcsStorage> getResources() {
+    return resources;
+  }
+
+  @Override
+  public String toString() {
+    return new StringBuilder("StorageRequestSummary(actionType=")
+        .append(actionType)
+        .append(", resources=[")
+        .append(resources.stream().map(GcsStorage::toString).collect(Collectors.joining(", ")))
+        .append("])")
+        .toString();
+  }
+
+  /** Encapsulation of GCS storage resource */
+  public static class GcsStorage {
+
+    private final String bucket;
+    private final String object;
+    private final boolean isSource;
+
+    public GcsStorage(String bucket, String object, boolean isSource) {
+      this.bucket = bucket;
+      this.object = object == null ? "/" : object;
+      this.isSource = isSource;
+    }
+
+    public String getBucket() {
+      return bucket;
+    }
+
+    public String getObject() {
+      return object;
+    }
+
+    public boolean isSource() {
+      return isSource;
+    }
+
+    public boolean isDestination() {
+      return !isSource;
+    }
```

Comment:
What does these methods imply for authorization?

Reply:
Some actions, like copy, for example, have two resources: one of them is the source and the other is destination. These methods is the only way to distinguish them.

Reply:
Maybe it's better to use two separate lists instead of one.